test/python/pulse/test_parameter_manager.py

```diff
@@ -0,0 +1,337 @@
+
+    def test_set_parameter_to_inst(self):
+        """Test get parameters from instruction."""
+        test_obj = pulse.ShiftPhase(self.phi1 + self.phi2, pulse.DriveChannel(0))
+
+        value_dict = {self.phi1: 0.123, self.phi2: 0.456}
+
+        visitor = ParameterSetter(param_map=value_dict)
+        assigned = visitor.visit(test_obj)
+
+        ref_obj = pulse.ShiftPhase(0.579, pulse.DriveChannel(0))
+
+        self.assertEqual(assigned, ref_obj)
+
+    def test_set_parameter_to_call(self):
+        """Test get parameters from instruction."""
+        sched = pulse.Schedule()
+        sched += pulse.ShiftPhase(self.phi1, self.d1)
+
+        test_obj = pulse.Call(subroutine=sched)
+
+        value_dict = {self.phi1: 1.57, self.ch1: 2}
+
+        visitor = ParameterSetter(param_map=value_dict)
+        assigned = visitor.visit(test_obj)
+
+        ref_sched = pulse.Schedule()
+        ref_sched += pulse.ShiftPhase(1.57, pulse.DriveChannel(2))
+
+        ref_obj = pulse.Call(subroutine=ref_sched)
+
+        self.assertEqual(assigned, ref_obj)
+
+    def test_set_parameter_to_alignment_context(self):
+        """Test get parameters from alignment context."""
+        test_obj = AlignEquispaced(duration=self.context_dur + self.dur1)
+
+        value_dict = {self.context_dur: 1000, self.dur1: 100}
+
+        visitor = ParameterSetter(param_map=value_dict)
+        assigned = visitor.visit(test_obj)
+
+        ref_obj = AlignEquispaced(duration=1100)
+
+        self.assertEqual(assigned, ref_obj)
+
+    def test_nested_assigment_partial_bind(self):
+        """Test nested schedule with call instruction.
+        Inline the schedule and partially bind parameters."""
+        context = AlignEquispaced(duration=self.context_dur)
+        subroutine = pulse.ScheduleBlock(alignment_context=context)
+        subroutine += pulse.Play(self.parametric_waveform1, self.d1)
+
+        nested_block = pulse.ScheduleBlock()
+        nested_block += pulse.Call(subroutine=subroutine)
+
+        test_obj = pulse.ScheduleBlock()
+        test_obj += nested_block
+
+        test_obj = inline_subroutines(test_obj)
+
+        value_dict = {self.context_dur: 1000, self.dur1: 200, self.ch1: 1}
+
+        visitor = ParameterSetter(param_map=value_dict)
+        assigned = visitor.visit(test_obj)
+
+        ref_context = AlignEquispaced(duration=1000)
+        ref_subroutine = pulse.ScheduleBlock(alignment_context=ref_context)
+        ref_subroutine += pulse.Play(pulse.Gaussian(200, self.amp1_1 + self.amp1_2, 25),
+                                     pulse.DriveChannel(1))
+
+        ref_nested_block = pulse.ScheduleBlock()
+        ref_nested_block += ref_subroutine
+
+        ref_obj = pulse.ScheduleBlock()
+        ref_obj += nested_block
+
+        self.assertEqual(assigned, ref_obj)
+
+    def test_set_parameter_to_complex_schedule(self):
+        """Test get parameters from complicated schedule."""
+        test_block = self.create_fully_parametric_schedule()
```

Comment:
It's very hard for me as the reviewer to see if the test output is as expected. Just a comment, it's okay if it's left as is. I'm trusting that you have double checked it

Reply:
This is just to avoid deepcopy (since parametrized schedule is updated with parameters). No special reason. I can remove this function.